tests/test_notify_v2.py

```diff
@@ -0,0 +1,415 @@
+        return json.loads(obj)
+
+    @testmode.standalone
+    def test_subscription_enumerate(self, replica=Replica.aws):
+        url = str(UrlBuilder()
+                  .set(path="/v1/subscriptions")
+                  .add_query("replica", replica.name)
+                  .add_query("subscription_type", "jmespath"))
+        self.assertGetResponse(
+            url,
+            requests.codes.ok,
+            headers=get_auth_header())
+
+    @testmode.integration
+    def test_get_subscriptions_for_replica(self):
+        for replica in Replica:
+            with self.subTest(replica.name):
+                subscriptions_v2.get_subscriptions_for_replica(replica)
+
+    @testmode.integration
+    def test_subscription_api(self):
+        for replica in Replica:
+            with self.subTest(replica.name):
+                self._test_subscription_api(replica)
+
+    def _test_subscription_api(self, replica: Replica):
+        with self.subTest(f"{replica}, Should not be able to PUT with invalid JMESPath"):
+            doc = {
+                'callback_url': "https://example.com",
+                'method': "POST",
+                'encoding': "application/json",
+                'form_fields': {'foo': "bar"},
+                'payload_form_field': "baz",
+                'jmespath_query': "not valid JMESPath",
+            }
+            self._put_subscription(doc, replica, codes=requests.codes.unprocessable)
+
+        with self.subTest(f"{replica}, PUT should succceed for valid JMESPath"):
+            sub = {
+                'callback_url': "https://example.com",
+                'method': "POST",
+                'encoding': "application/json",
+                'form_fields': {'foo': "bar"},
+                'payload_form_field': "baz",
+                'jmespath_query': "foo",
+            }
+            self._put_subscription(sub, replica)
+
+        with self.subTest(f"{replica}, PUT should succceed for missing JMESPath"):
+            sub = {
+                'callback_url': "https://example.com",
+                'method': "POST",
+                'encoding': "application/json",
+                'form_fields': {'foo': "bar"},
+                'payload_form_field': "baz",
+            }
+            subscription = self._put_subscription(sub, replica)
+
+        with self.subTest(f"{replica}, GET should succceed"):
+            self._get_subscription(subscription['uuid'], replica)
+
+        with self.subTest(f"{replica}, DELETE should succceed"):
+            self._delete_subscription(subscription['uuid'], replica)
+
+        with self.subTest(f"{replica}, DELETE on non-existent subscription shoudld return not-found"):
+            self._delete_subscription(str(uuid4()), replica, codes=404)
+
+    @testmode.standalone
+    def test_should_notify(self):
+        for replica in Replica:
+            with self.subTest(replica.name):
+                self._test_should_notify(replica)
+
+    def _test_should_notify(self, replica):
```

Comment:
from the descriptions of your subtest it's not clear what your testing.

Reply:
are these negative test cases?

Reply:
Yes, documentation improved